presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -363,6 +364,17 @@ public ConnectorPageSource getPageSource()
         if (fromType instanceof VarcharType && (toHiveType.equals(HIVE_BYTE) || toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG))) {
             return new VarcharToIntegerNumberCoercer<>((VarcharType) fromType, toType);
         }
+        if (fromType instanceof VarcharType && toType instanceof VarcharType) {
+            VarcharType toVarcharType = (VarcharType) toType;
+            VarcharType fromVarcharType = (VarcharType) fromType;
+
+            if (narrowerThan(toVarcharType, fromVarcharType)) {
+                return new VarcharCoercer(fromVarcharType, toVarcharType);
+            }
+
+            // Coercion not worth the copying penalty when toType is wider
```

Comment:
"not worth" is not the right term. "is type only" would be more accurate, but i would actually remove the comment.